### PR TITLE
Dev/xygu/20220429/localization

### PR DESCRIPTION
**Pull request by Xiaoy312**

### CODE REVIEW

**Review by nickrandolph — APPROVED** [review]
I've made a couple of updates to this branch - please make sure you get a fresh copy, as I rebased of main.
This branch is good to merge but localization will need an update from Uno.Extensions before changes to localization are shown correctly after a restart.

**Review by francoistanguay — DISMISSED** [review]

**Review by agneszitte-nventive — APPROVED** [review]

**Comment by agneszitte-nventive** [comment]
Related Issue @Xiaoy312 : https://github.com/unoplatform/uno.todo/issues/148

File: src/ToDo.UI/Views/SettingsPage.xaml [hunk 1]
```diff
@@ -41,6 +41,7 @@
 						   Text="{Binding CurrentUser.Email}"
 						   Style="{StaticResource MaterialBody2}" />
 				<Button Content="sign out"
+						x:Uid="SignOutButton"
```
Comment (agneszitte-nventive) [inline src/ToDo.UI/Views/SettingsPage.xaml:44]: Can we update the x:Uid of this page to match the other please @Xiaoy312 ? (x:Uid="**NameOfThePage**_Name")
@carldebilly I don't think the generated code for the plugin already follow the same pattern as the x:Uid in this PR no ? Should we open a figma issue for it ?

Comment (Xiaoy312) [reply src/ToDo.UI/Views/SettingsPage.xaml:44]: This PR just cover some of the pages. I havent gone through SettingsPage yet.
The plan is to merge this ASAP, so I dont have to deal with further merging conflicts...

Comment (Xiaoy312) [reply src/ToDo.UI/Views/SettingsPage.xaml:44]: And, then I will make the rest in another PR or PRs

Comment (agneszitte-nventive) [reply src/ToDo.UI/Views/SettingsPage.xaml:44]: Ok no problem and I understand ;) go ahead and @carldebilly feel free to give us your opinion regarding the generated code.

**Review by agneszitte-nventive — APPROVED** [review]

**Review by agneszitte-nventive — APPROVED** [review]

File: src/ToDo.UI/Views/SettingsPage.xaml (hunk 1 above)
Comment (carldebilly) [reply src/ToDo.UI/Views/SettingsPage.xaml:44]: @agneszitte-nventive you're right it should follow the generated format. We'll need to make sure the names are correct in Figma, but for now, that's correct.

File: src/ToDo.UI/Views/SettingsPage.xaml.cs [hunk 2]
```diff
@@ -1,39 +1,81 @@
...
+		if (ViewModel is null)
+		{
+			return;
+		}
+
+		isInitializing = true;
+
+		// Set default theme
+		var currentTheme = SystemThemeHelper.IsRootInDarkMode(XamlRoot) ? "Dark" : "Light";
+		SelectChipGroupItem(ThemeChipGroup, x => (string)x.Tag == currentTheme);
```
Comment (carldebilly) [inline src/ToDo.UI/Views/SettingsPage.xaml.cs:43]: Something is clearly missing.

**Pull request merged**